src/Microsoft.Health.Fhir.Liquid.Converter/Filters/DateFilters.cs

```diff
@@ -96,6 +98,31 @@ public static string Now(string input, string format = "yyyy-MM-ddTHH:mm:ss.FFFZ
             return DateTime.UtcNow.ToString(format);
         }
 
+        /* Converts an HL7v2 time interval to the FHIR format for time
+        /    HL7v2 - 2.8.35.2 Explicit time interval (ST)
+        /    FHIR  - time https://build.fhir.org/datatypes.html#time
+        */
+        public static string FormatTimeWithColon(string input, string inputFormat = "HHmm")
+        {
+            // For backwards compatibility, if the input is not a 4 digit numeric value then just return the value
+            if (string.IsNullOrEmpty(input) || (!(input.All(char.IsDigit) && (input.Length == 4))))
```

Comment:
@shabanlushaj agree - it is better to make this more generic for the other HL7v2 times. I've updated the pull request so that the filter is more generic. 

Question: Is `HHmmssSS` supported by HL7v2 for TM time? I was looking at the spec and it seemed to be `HHMM[SS[.SSSS]][+/-ZZZZ]` with the milliseconds following a decimal. But either way both formats HHmmssff and HHmmss.ff should be supported now.

Reply:
@wi-y Whenever it is UTC, timezone can be discarded. I believe same applies for time. It would be nice to have this filter support timezone.

`20240107111232-0300` for datetime this is supported by converter (MSH.7 - Date/Time Of Message).

Tried something that might do the work, but we have to ignore check for input length and inputFormat having same length.
```cs
var input = "144100.0000+0300";
var inputFormat = "HHmmss.ffffzzzz";
var dt = DateTime.ParseExact(input, inputFormat, CultureInfo.InvariantCulture).TimeOfDay;
```